source/slang/hlsl.meta.slang

```diff
@@ -19889,3 +19889,50 @@ struct ConstBufferPointer
         }
     }
 }
+
+//
+// HLSL-like dynamic resources
+// https://microsoft.github.io/DirectX-Specs/d3d/HLSL_SM_6_6_DynamicResources.html
+//
+// For Khronos targets, `__DynamicResource` can be used to declare "untyped" global bindings as
+// usual (e.g. unsized arrays for descriptor indexing), which will then be materialized into 
+// new aliased bindings for each distinct cast type.
+//
+
+__magic_type(DynamicResourceType)
+__intrinsic_type($(kIROp_DynamicResourceType))
+struct __DynamicResource<let kind = __DynamicResourceKind.General>
+{
+    __intrinsic_op($(kIROp_CastDynamicResource))
```

Comment:
This is a downcast right?

Reply:
I mean we are semantically treating all texture and buffer types as subtypes that conforms to IDynamicCastable, so casting from DynamicResource to a Texture is more of a downcast(casting from a generic to a more derived type) than a up cast.

Reply:
Yes I think from a semantic perspective that makes sense.